clients/src/main/java/org/apache/kafka/clients/consumer/NoOffsetForPartitionException.java

```diff
@@ -26,19 +26,20 @@
 
     private static final long serialVersionUID = 1L;
 
-    private final TopicPartition partition;
+    private final Set<TopicPartition> partitionSet;
 
-    public NoOffsetForPartitionException(TopicPartition partition) {
-        super("Undefined offset with no reset policy for partition: " + partition);
-        this.partition = partition;
+    public NoOffsetForPartitionException(String message) {
```

Comment:
I had avoided this constructor intentionally so that `partitions()` could be depended on in user code. There are a couple of similar cases. Seems like we need a better convention to allow exceptions to include extra information.